runtimes/talos-libgdx/src/com/talosvfx/talos/runtime/values/NumericalValue.java

```diff
@@ -149,6 +149,22 @@ public void set(NumericalValue val1, NumericalValue val2, NumericalValue val3) {
 		elements[1] = val2.getFloat();
 		elements[2] = val3.getFloat();
 	}
+	public void set(NumericalValue val1, NumericalValue val2, NumericalValue val3,NumericalValue val4) {
+		currentElementCount = 4;
+		elements[0] = val1.getFloat();
+		elements[1] = val2.getFloat();
+		elements[2] = val3.getFloat();
+		elements[4] = val4.getFloat();
+	}
+	public void set(NumericalValue ... val) {
```

Comment:
Varargs should be avoided as they generate garbage behind the scenes. (Which is why we have the multiple set methods.

Reply:
Thanks for the explanation